test/Sentry.Unity.Tests/IntegrationTests.cs

```diff
@@ -239,6 +240,65 @@ public IEnumerator BugFarmScene_DebugLogException_IsMarkedUnhandled()
                     exception.Mechanism?.Handled is false) is not null));
         }
 
+        [UnityTest]
+        public IEnumerator BugFarmScene_DebugLogError_IsCaptured()
+        {
+            yield return BugFarmScene_DebugLog(inTask: false, logException: false);
+        }
+
+        [UnityTest]
+        public IEnumerator BugFarmScene_DebugLogError_IsCapturedInTask()
+        {
+            yield return BugFarmScene_DebugLog(inTask: true, logException: false);
+        }
+        [UnityTest]
+        public IEnumerator BugFarmScene_DebugLogException_IsCaptured()
+        {
+            yield return BugFarmScene_DebugLog(inTask: false, logException: true);
+        }
+
+        [UnityTest]
+        public IEnumerator BugFarmScene_DebugLogException_IsCapturedInTask()
+        {
+            yield return BugFarmScene_DebugLog(inTask: true, logException: true);
+        }
+
+        // Note: can't use nunit TestCase() because it's not supported with IEnumerator return.
+        private IEnumerator BugFarmScene_DebugLog(bool inTask, bool logException)
+        {
+            yield return SetupSceneCoroutine("1_BugFarm");
+
+            // arrange
+            var testEventCapture = new TestEventCapture();
+            using var _ = InitSentrySdk(o =>
+            {
+                o.AddIntegration(new UnityApplicationLoggingIntegration(eventCapture: testEventCapture));
+            });
+            var testBehaviour = new GameObject("TestHolder").AddComponent<TestMonoBehaviour>();
+
+            // act
+            var method = logException
+                ? (inTask ? nameof(testBehaviour.DebugLogExceptionInTask) : nameof(testBehaviour.DebugLogException))
+                : (inTask ? nameof(testBehaviour.DebugLogErrorInTask) : nameof(testBehaviour.DebugLogError));
+            testBehaviour.gameObject.SendMessage(method);
+
+            // wait
+            if (inTask)
+            {
+                var stopWatch = new Stopwatch();
+                stopWatch.Start();
+                while (testEventCapture.Events.Count == 0 && stopWatch.ElapsedMilliseconds < 5000)
+                {
+                    yield return null;
+                }
+            }
+
+            // assert
+            Assert.AreEqual(1, testEventCapture.Events.Count);
+            var isMainThread = testEventCapture.Events.First().Tags.SingleOrDefault(t => t.Key == "unity.is_main_thread");
+            Assert.AreEqual((!inTask).ToString(), isMainThread.Value);
```

Comment:
It seems flakey? 
https://github.com/getsentry/sentry-unity/actions/runs/2115833801/attempts/1
https://github.com/getsentry/sentry-unity/runs/5968904127?check_suite_focus=true

Reply:
Has this: https://github.com/getsentry/sentry-unity/issues/251 finally caught up with us?

Reply:
fixed by making sure SDK Dispose() is being called before the test finishes, so that there are no outstanding envelopes

Reply:
OK, not completely fixed then... https://github.com/getsentry/sentry-unity/runs/5990231610?check_suite_focus=true

Reply:
Can't reproduce that locally, but I'm getting other errors similar to this:

```
Unhandled log message: '[Error] Sentry: (Error) Exception in background worker of CachingTransport. System.IO.IOException: Sharing violation on path C:/Users/dlugo/AppData/LocalLow/DefaultCompany/unity-of-bugs\Sentry\618762E37120B6E01219C43B9076FFEC1ACE8406\__processing\1649773679_3939__1948942592.envelope
  at System.IO.File.Delete (System.String path) [0x00073] in <eae584ce26bc40229c1b1aa476bfa589>:0 
  at Sentry.Internal.Http.CachingTransport+<InnerProcessCacheAsync>d__17.MoveNext () [0x002a9] in <e46e07d242f048179999156e39b482f9>:0 
```

Must be something related to initializing the .net SDK multiple times during the same runtime. I know the .net SDK isn't really supposed to be used like that, because we had to specifically implement workarounds to NOT initialize it multiple times on Windows for example. 

I'm thinking we should split out these unit tests out of the example app, so they're not subject to the same automatic initialization via `SentryInitialization.cs`. Alternatively, we can just configure Sentry to be disabled before running these tests.

Reply:
I've removed the DSN from options in CI just before the PlayerMode test is run, so that SentryInitialization.init() doesn't run and we only run the manual init in the tests. Tests seem to be fixed now.